lib/daru/vector.rb

```diff
@@ -352,7 +352,14 @@ def == other
         if other.is_a?(Daru::Vector)
           mod.apply_vector_operator operator, self, other
         else
-          mod.apply_scalar_operator operator, @data,other
+          missing_undefined_indexes = nil_positions + nan_positions
+          if operator != :== && operator != :!= && missing_undefined_indexes.size
+            data_dup = @data.dup
+            missing_undefined_indexes.each { |i| data_dup.delete_at(i) }
+            mod.apply_scalar_operator operator, data_dup, other
+          else
+            mod.apply_scalar_operator operator, data, other
+          end
```

Comment:
I don't feel it as a readable code. 
First, what is the point? "When we are _applying scalar operator_,  we should ignore 'missing' values", right? (And only for _some_ of operators.) Why don't you think it is a responsibility of `apply_scalar_operator` itself to tell applicable elements from others, and the operators that are ok with "missing values" from those that are not? (And consider, BTW, the fact that `Float::NAN == Float::NAN # => false`)